closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurityConfigurator.java

```diff
@@ -0,0 +1,321 @@
+/*
+ * ===========================================================================
+ * (c) Copyright IBM Corp. 2022, 2022 All Rights Reserved
+ * ===========================================================================
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * IBM designates this particular file as subject to the "Classpath" exception
+ * as provided by IBM in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, see <http://www.gnu.org/licenses/>.
+ *
+ * ===========================================================================
+ */
+
+package openj9.internal.security;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Properties;
+
+import sun.security.util.Debug;
+
+/**
+ * Configures the security providers when in restricted security mode.
+ */
+public final class RestrictedSecurityConfigurator {
+
+    private static final Debug debug = Debug.getInstance("semerufips");
+
+    // Restricted security mode enable check, only supported on Linux x64.
+    private static final boolean userEnabledSecurity;
+    private static final boolean isSecuritySupported;
+    private static final boolean shouldEnableSecurity;
+    private static final String userSecuritySetting;
+    private static final boolean userEnabledFIPS;
+
+    private static String userSecurityNum = "0";
```

Comment:
I don't think that's sufficient motivation to use `String` instead of `int`. The difference would just be use of `StringBuilder.append(int)` instead of `StringBuilder.append(String)`; if we were concerned about performance here, we might extract the common subexpression `"RestrictedSecurity" + userSecurityNum + ".desc`.

Reply:
@keithc-ca 
Thanks for the advice, made the changes and will check in the code later with other changes.